packages/lesswrong/components/ea-forum/wrapped/EAForumWrapped2023Page.tsx

```diff
@@ -0,0 +1,1501 @@
+    <p className={classNames(classes.summaryLinkWrapper, classes.text, classes.mt70)}>
+      Here’s your 2023 all in one page
+      <Components.ForumIcon icon="NarrowArrowDown" />
+    </p>
+  </section>
+}
+
+/**
+ * This is the primary page component for EA Forum Wrapped 2023.
+ */
+const EAForumWrapped2023Page = ({classes}: {classes: ClassesType}) => {
+  const currentUser = useCurrentUser()
+
+  const { data } = useForumWrappedV2({
+    userId: currentUser?._id,
+    year: 2023
+  })
+  
+  useEffect(() => {
+    if (currentUser) {
+      // Add scrollSnapType: 'y mandatory' styles to html and body elements (need both for different browsers?)
+      document.body.classList.add('EAForumWrapped2023Page-scrollSnap')
+      document.documentElement.classList.add('EAForumWrapped2023Page-scrollSnap')
+    }
+  }, [currentUser])
+
+  const { HeadTags, ForumIcon, CloudinaryImage2, UsersProfileImage, CoreTagIcon } = Components
+  
+  // If there's no logged in user, prompt them to login
+  if (!currentUser) {
+    return <AnalyticsContext pageContext="eaYearWrapped">
+      <HeadTags
+        title="EA Forum 2023 Wrapped"
+        image={makeCloudinaryImageUrl('b90e48ae75ae9f3d73bbcf17f2ddf6a0', socialImageProps)}
+      />
+      <main className={classes.root}>
+        <section className={classes.section}>
+          <h1 className={classes.heading1}>Your 2023 Wrapped</h1>
+          <div className={classes.loginWrapper}>
+            <p className={classes.loginText}>
+              <a href={`/auth/auth0?returnTo=${encodeURIComponent('/wrapped')}`} className={classes.link}>
+                Login
+              </a>{" "}
+              to see your 2023 data, or{" "}
+              <a href={`/auth/auth0?screen_hint=signup&returnTo=${encodeURIComponent('/wrapped')}`} className={classes.link}>
+                Sign up
+              </a>{" "}
+              to get ready for 2024
+            </p>
+          </div>
+          <CloudinaryImage2 publicId="b90e48ae75ae9f3d73bbcf17f2ddf6a0" wrapperClassName={classes.loginImgWrapper} className={classes.img} />
+        </section>
+      </main>
+    </AnalyticsContext>
+  }
+
+  if (!data) return null;
+
+  // TODO un-admin gate
+  if (!userIsAdminOrMod(currentUser)) {
+    return <div className={classes.root}>
+      You do not have permission to view this page.
+    </div>
+  }
+
+  return (
+    <AnalyticsContext pageContext="eaYearWrapped">
+      <main className={classes.root}>
+        
+        <section className={classes.section}>
+          <h1 className={classes.heading1}>Your 2023 Wrapped</h1>
+          {/* TODO: finish building this
+          <button className={classNames(classes.summaryLinkWrapper, classes.skipToSummaryBtn)}>
+            Skip to summary
+            <ForumIcon icon="NarrowArrowDown" />
+          </button> */}
+          <CloudinaryImage2 publicId="b90e48ae75ae9f3d73bbcf17f2ddf6a0" wrapperClassName={classes.imgWrapper} className={classes.img} />
+        </section>
```

Comment:
Sorry this file is so gigantic! I split up the sections into separate components so hopefully that makes this easier to read. I didn't do the first and last sections yet because we worked on them a bit on the other branch, but I will turn them into separate components in the next PR.